frontend/src/hooks/useWidgetPaging.js

```diff
@@ -126,6 +126,21 @@ export default function useWidgetPaging(
         dataToExport = dataToUse.filter((row) => selectedRowsIds.includes(row.id));
       }
 
+      // Create data array if its missing (headers array must be populated correctly).
+      const createMissingData = dataToExport.every((d) => !d.data);
+      if (createMissingData) {
+        dataToExport = dataToExport.map((d) => ({
+          ...d,
+          heading: d.name,
+          data: Object.keys(d)
+            .filter((key) => headers.includes(key))
+            .map((key) => ({
+              title: key,
+              value: d[key].toString().replace(/,/g, ''),
+            })),
+        }));
+      }
+
```

Comment:
can you explain in a comment why we need this? I don't get it by looking at it.

Reply:
Yea this probably will go away. I was attempting to re-use the export to CSV for a different underlying set. But given the fact that this data displays totally differently I probably will write a separate export to CSV (or handle this on the BE).

In short this code will probably go away before being merged into main.